Libraries/NavigationExperimental/NavigationTypeDefinition.js

```diff
@@ -111,4 +113,25 @@ export type NavigationStyleInterpolator = (
   props: NavigationSceneRendererProps,
 ) => Object;
 
-export type NavigationTransitionConfigurator = () => NavigationTransitionSpec;
+export type NavigationTransitionConfigurator = (
+  transitionProps: NavigationTransitionProps,
+  prevTransitionProps: ?NavigationTransitionProps,
+) => NavigationTransitionSpec;
+
+export type AnimationConfig = Object;
+
+export type AnimationEndResult = {
+  finished: bool
+};
+
+export type AnimationEndCallback = (result: AnimationEndResult) => void;
+
+export type CompositeAnimation = {
+  start: (callback?: ?AnimationEndCallback) => void;
+  stop: () => void;
+};
+
+export type NavigationTransitionTiming = (
+  value: NavigationAnimatedValue,
+  config: AnimationConfig,
+) => CompositeAnimation;
```

Comment:
Having  type definition for animation that we don;t really own in Navigation seems strange. 

```
export type NavigationTransitionTiming = (
   value: NavigationAnimatedValue,
   config: Object,
) => Object,
```

I think this should be good enough.